charm_versions.md

```diff
@@ -0,0 +1,133 @@
+is provided.
+
+To upgrade your local Juju version, use the same tool you used to install it. For example, if you
+installed Juju with snap, you can run `snap refresh juju --channel=3.3/stable` to upgrade to
+Juju 3.3.
+
+## Charms
+
+To upgrade the version or revision of an installed charm,
+[use `juju refresh`](https://juju.is/docs/juju/manage-charms#heading--update-a-charm) for both
+local and Charmhub charms.
+
+### Charm Version
+
+Charms can optionally specify a version of the charm itself. You can find this information with
+`juju status` in the YAML or JSON formats in the `applications.<app name>.charm-version` field. If
+there is no version, the key will not be present in the status output.
+
+The version is an arbitrary string, so could be a major.minor.patch style version, a version control
+hash identifier, or anything else.
+
+To set the version, charms include a file called `version` (no extension) at the top level of the
+charm (the same folder as the `charmcraft.yaml` file), where the content of the file is the version
+string.
+
+In the past, the version would be set automatically if the charm was deployed from a known version
+control system, but that is no longer the case. The intention is that charmers set this version to
+allow tracking the installed version of the charm back to the source tree that it was built from.
+
+### Charm Revision
+
+All charms have a revision, which is a simple positive integer. You can see the revisions available
+of a charm published to Charmhub with `charmcraft revisions <charm name>`, and in the drop-down near
+the top of a charm's page in Charmhub. You can see the installed revision of a charm in
+`juju status` in the application table in the "Rev" column in the tabular view, or in the
+`applications.<app name>.charm-rev` field in JSON or YAML format.
+
+The revision is automatically set when uploading the charm using `charmcraft upload`. Note that
+revisions may be unreleased, or may be released only for specific
+[channels](https://juju.is/docs/sdk/charmcraft-release).
+
+When a charm is deployed locally, Juju will use an initial revision of 1 and then increment the
+revision each time a new version of the same charm is deployed or refreshed.
+
+### Libraries
+
+Charm libs have both a "API version" and a "patch version". Any changes to the lib API that are not
+backwards compatible will be made in a new API version, while new patch versions include compatible
+improvements including new features, general improvements, and bug fixes.
+
+The available versions for a charm lib can be found on Charmhub in the "Libraries" tab. These can
+also be found using `charmcraft list-lib <charm name>`.
+
+The API and patch versions of a charm lib are contained within the lib source file itself. Search
+for `LIBAPI` (the API version) and `LIBPATCH` (the patch version) within the source`.
+
+It is not possible to see which version of a lib was used in a deployed charm, other than by using
```

Comment:
This is not quite true -- there isn't an off-the-shelf way, but I imagine you could fairly easily ssh in and use `grep LIBAPI libfile.py` to figure this out. May or may not want to mention this.

Reply:
Fair call; adjusted a bit.